Documentation/policy-comments.md

```diff
@@ -0,0 +1,169 @@
+no comments, you can only guess what was the real intent of the author.
+
+The design of the code and the intentions of its author are important knowledge and the knowledge has to be shared. 
+Otherwise, it is hard for everyone to work with the code if the only person with that knowledge is not available.
+Spend the time now to add comments to save the time to your collegues, or your future self, later. If you do that, you will 
+find out that in order to create a good comment, you have to understand what you are doing. By trying to explain your intention, 
+you will often find a problem in your design or its implementation and you will actually be able to complete your task faster.
+
+If you are interested to learn more about why to comment your code, there are almost unlimited number of resources on this 
+topic on the Internet. You will find many blogs and even whole books talking about the necessity of comments.
+
+
+## Rules
+
+We have two types of rules. Hard rules are those that are required and your code will not be accepted if it violates 
+them. Soft rules are more like guidelines - try to use them where possible. The reviewer can ask you to update your code 
+if it violates a soft rule and it can be fixed easily.
+
+ * **XMLDOC** - Use triple slash (XML documentation) comments for classes, interfaces, methods, parameters, fields, properties, structures, enums, etc. These comments are required - even for trivial code. *(HARD)*
+   * **SUMMARY-1** - First paragraph in `<summary>` must be short, no more than 3 lines. If you find you need more than 2 lines, maybe you are not writing the summary anymore and you are already explaining details. *(HARD)*
+   * **SUMMARY-2** - `<summary>` should contain information useful for someone who is trying to understand the code. Implementation details belong to `<remarks>`. *(SOFT)*
+   * **METHODS** - Method comments must contain `<summary>`. If the method has input and/or output parameters `<param>` must be used for each of them. If the method returns anything else than `void` or `Task` (but not `Task<T>`),
+it must contain `<returns>`. The order of the tags is `<summary>`, `<param>`, `<returns>`, `<exception>`, `<remarks>`, `<example>`, `<seealso>`. *(HARD)*
+   * **FORMAT** - Use `<para>`, `<see>`, `<seealso>`, `<c>`, `<code>`, `<list>` to make the comment structure better. *(SOFT)*
+   * **INHERITDOC** - Use `<inheritdoc />` if the comment already exists on lower level (base class, interface, virtual/abstract method). But feel free to add additional comments if the base comment is not sufficient. *(HARD)*
+ * **INLINE** - Use double slash (inline) comments for function body code, variables, and inline comments. *(HARD)*
+   * **USE** - Double slash comments are optional and should be used to explain complicated or counter-intuitive segments of the code. *(SOFT)*
+   * **TRIVIAL** - Do not use double slash comments for trivial constructs. *(HARD)*
```

Comment:
Double slash trivial is something like this:
```
public int CalculateThat(int a, int b)
{
  // Calculates the sum of a and b.
  int x = a + b;
  // Returns x raised to the power of 2.
  return x * x;
}
```

Those comments strictly describe what the code does. They are pure translation of C# to English language. Thus they are trivial. Those comments are forbidden.